service/docs/source/overview.rst

```diff
@@ -1,14 +1,758 @@
+
+        #include <stdio.h>
+        #include <geopm_topo.h>
+        #include <geopm_pio.h>
+        #include <geopm_error.h>
+
+        int main (int argc, char** argv)
+        {
+            char err_msg[1024];
+
+            int err = geopm_pio_write_control("CPU_FREQUENCY_MAX_CONTROL",
+                                              GEOPM_DOMAIN_CORE,
+                                              0,
+                                              3.0e9);
+
+            if (err != 0) {
+                geopm_error_message(err, err_msg, 1024);
+                printf("Err msg = %s\n", err_msg);
+            }
+
+            return 0;
+        }
+
+    .. code-tab:: c++
+
+        // Write the current CPU frequency for core 0 to 3.0 GHz
+
+        #include <iostream>
+        #include <geopm/PlatformIO.hpp>
+        #include <geopm/PlatformTopo.hpp>
+
+        int main (int argc, char** argv)
+        {
+            geopm::platform_io().write_control("CPU_FREQUENCY_MAX_CONTROL",
+                                               GEOPM_DOMAIN_CORE, 0,
+                                               3.0e9);
+
+            return 0;
+        }
+
+    .. code-tab:: python
+
+        # Write the current CPU frequency for core 0 to 3.0 GHz
+
+        import geopmdpy.topo as topo
+        import geopmdpy.pio as pio
+
+        pio.write_control('CPU_FREQUENCY_MAX_CONTROL', topo.DOMAIN_CORE, 0, 3.0e9)
+
+.. note::
+
+    To determine the initial value of any control, use ``geopmread`` or the
+    corresponding ``PlatformIO`` APIs at the desired domain.  E.g.:
+
+    .. code-block:: bash
+
+        $ geopmread CPU_FREQUENCY_MAX_CONTROL core 0
+
+Understanding Disaggregation
+""""""""""""""""""""""""""""
+
+Just as signals can be aggregated to a more coarse domain from their native
+one, controls can be disaggregated from a coarse domain to their native domain.
+This happens automatically with ``geopmwrite`` and the corresponding APIs.
+
+Using ``CPU_FREQUENCY_MAX_CONTROL`` as an example, the output in `Listing Control
+Information`_ shows the native domain as ``core``.  If the user desires to
+write the same value to all the cores in a package, simply issue the request at
+the ``package`` domain, and all cores in that package will be written.
+Simiarly, to write the same value to all cores in all packages, issue the
+request at the ``board`` domain.
+
+To determine how the disaggregation will occur for a given control, you need to
+examine the aggregation type.  In this example, ``CPU_FREQUENCY_MAX_CONTROL`` has an
+aggregation type of ``expect_same``.  When writing this control at a more
+coarse domain than the native one, this means that all the native domains will
+receive the same value as the coarse domain.
+
+Controls that use any other aggregation type will have the requested value
+distributed evenly across the native domain.  Taking
+``MSR::PKG_POWER_LIMIT:PL1_POWER_LIMIT`` as an example, it has the following
+information:
```

Comment:
This is not true.  The only aggregation type that determines that the requested value should be divided by the number of native domains is the aggregation type sum.  All other types will write the requested value without modification.